npm-install-peer-dependencies

```diff
@@ -55,6 +55,10 @@ node -e " \
               continue
         }
 
+        # cat <<< "$(jq ".devDependencies += {\"${NAME}\": \"${VSN}\"}" < package.json)" > package.json
+        cat <<< "$(npx json "this.devDependencies = this.devDependencies || {}" < package.json)" > package.json
```

Comment:
I can guess what it does, but you gonna need to explain to me how it actually works 😄

Reply:
here goes some "bashsplaining" :joy:

if you simplify the command : `cat <<< "$(foo < bar)" > bar`, you have

* `> bar` - write stdout to file
* `< bar` - read stdin from file
* `cat` - is cat
* `<<< ""` - is a oneline heredoc into stdin

in plain english:
 - output to stdout (via cat) the contents of the heredoc, and the heredoc is actually a command, and then into bar
- the key is the heredoc, because it doesn't stream, thus you buffer the command output, and only after you process the entire `package.json`, you start writing back to `package.json`

Reply:
Thank you a lot!